fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java

```diff
@@ -62,18 +62,68 @@ public static ReferenceValue createReferenceValueFrom(Reference ref, String base
         Integer version = null;
 
         if (ref.getReference() != null && ref.getReference().getValue() != null) {
+            return createReferenceValueFrom(ref.getReference().getValue(), ref.getType() != null ? ref.getType().getValue() : null, baseUrl);
+        } else if (ref.getIdentifier() != null && ref.getIdentifier().getValue() != null) {
+            // LOGICAL REFERENCE
+            value = ref.getIdentifier().getValue().getValue();
```

Comment:
In very very rare circumstances the .getValue can be null. 
Should there be a check on 66 to see it's null 

Example of null. 

```
"identifier": [
             ...
            "_value": {
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/StructureDefinition/data-absent-reason",
                        "valueCode": "unknown"
                    }
                ]
            }...
        }
    ]
```

You may want to double check me on the extension part.

Reply:
added the null check